src/main/java/ttt/CommandPrompt.java

```diff
@@ -67,44 +75,137 @@ public void print(Board board) {
 
     @Override
     public void printWinningMessageFor(PlayerSymbol symbol) {
-        display("Congratulations - " + symbol + " has won" + newLine());
+        display(FONT_COLOUR_ANSII_CHARACTERS
+                + "Congratulations - "
+                + colour(symbol)
+                + FONT_COLOUR_ANSII_CHARACTERS
+                + " has won");
     }
 
     @Override
     public void printDrawMessage() {
-        display("No winner this time" + newLine());
+        display(FONT_COLOUR_ANSII_CHARACTERS
+                + "No winner this time");
     }
 
-    @Override
-    public void clear() {
-        display(CLEAR_SCREEN_ANSII_CHARACTERS);
+    private Function<ValidationResult, Void> functionToRepromptGameType() {
+        return validationResult -> {
+            clear();
+            display(BOARD_COLOUR_ANSII_CHARACTERS + validationResult.reason());
+            askUserForGameType();
+            return null;
+        };
+    }
+
+    private Function<ValidationResult, Void> functionToRepromptReplay() {
+        return validationResult -> {
+            clear();
+            display(BOARD_COLOUR_ANSII_CHARACTERS + validationResult.reason());
+            askUserToPlayAgain();
+            return null;
+        };
+    }
+
+    private int asInteger(String input) {
+        return Integer.valueOf(input);
+    }
+
+    private Function<ValidationResult, Void> functionToRepromptForValidMove(Board currentBoard) {
+        return validationResult -> {
+            clear();
+            print(currentBoard);
+            display(validationResult.reason());
+            askUserForTheirMove();
+            return null;
+        };
+    }
+
+    private List<InputValidator> gameTypeValidators() {
+        return Arrays.asList(new NumericValidator(), new GameTypeValidator());
+    }
+
+    private String getValidInput(InputValidator compoundValidator, String input, Function reprompt) {
```

Comment:
Given that you are generifying the `Function` in all other methods you should possibly do the same here to avoid the warning message from the compiler (or IntelliJ):

```
  private String getValidInput(InputValidator compoundValidator, String input, 
                                              Function<ValidationResult, Void> reprompt) {
```